libs/metrics/cdc_transmission_levels.py

```diff
@@ -77,11 +81,41 @@ def overall_transmission_level(
     return TransmissionLevel.UNKNOWN
 
 
-def calculate_transmission_level_from_metrics(
-    metrics: can_api_v2_definition.Metrics,
+def calculate_transmission_level(
+    case_density: float, test_positivity_ratio: float
 ) -> can_api_v2_definition.CDCTransmissionLevel:
-    case_density_level = case_density_transmission_level(metrics.caseDensity)
-    test_positivity_level = test_positivity_transmission_level(metrics.testPositivityRatio)
+    case_density_level = case_density_transmission_level(case_density)
+    test_positivity_level = test_positivity_transmission_level(test_positivity_ratio)
 
     overall_level = overall_transmission_level(case_density_level, test_positivity_level)
     return overall_level
+
+
+def calculate_transmission_level_from_metrics(
+    metrics: can_api_v2_definition.Metrics,
+) -> can_api_v2_definition.CDCTransmissionLevel:
+    return calculate_transmission_level(metrics.caseDensity, metrics.testPositivityRatio)
+
+
+def calculate_cdc_transmission_level_from_row(row: pd.Series):
+    case_density = row[MetricsFields.CASE_DENSITY_RATIO]
+    test_positivity_ratio = row[MetricsFields.TEST_POSITIVITY]
+
+    return calculate_transmission_level(case_density, test_positivity_ratio)
+
+
+def calculate_transmission_level_timeseries(
+    metrics_df: pd.DataFrame, metric_max_lookback=top_level_metrics.MAX_METRIC_LOOKBACK_DAYS
+):
```

Comment:
just for clarity, in the case where there is neither test positivity nor case data for a specific day (but there is data within `MAX_METRIC_LOOKBACK_DAYS` of the day), this will forward propagate the most recent transmission level to this day, right?

Reply:
Yeah, that is correct.  I think things get a bit interesting when only one metric has data within MAX_METRIC_LOOKBACK_DAYS. Once one metric gets too stale, the transmission level will transition to only reflect the non-stale metric, so we don't end up with a strict forward-fill of the transmission level.

Let me know if you think this could use better commenting.

Reply:
Ok cool that makes sense. and no I think it's ok as is, just wanted to check!